imagetest/test_suites/network/alias_ip_test.go

```diff
@@ -0,0 +1,141 @@
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+const (
+	markerFile = "/boot-marker"
+)
+
+func TestAliasAfterOnBoot(t *testing.T) {
+	if err := verifyIPAliases(); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestAliasAfterReboot(t *testing.T) {
+	_, err := os.Stat(markerFile)
+	if os.IsNotExist(err) {
+		// first boot
+		if _, err := os.Create(markerFile); err != nil {
+			t.Fatalf("failed creating marker file: %v", err)
+		}
+		t.Fatal("first boot")
+	}
+	// second boot
+	if err := verifyIPAliases(); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func verifyIPAliases() error {
+	var networkInterface string
+
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		return fmt.Errorf("couldn't get image from metadata: %v", err)
+	}
+	switch {
+	case strings.Contains(image, "debian-10"), strings.Contains(image, "ubuntu"):
+		networkInterface = defaultPredictableInterface
+	default:
+		networkInterface = defaultInterface
+	}
+
+	actualIPs, err := getGoogleRoutes(networkInterface)
+	if err != nil {
+		return fmt.Errorf("failed get ip alises")
+	}
+	if err := verifyIPExist(actualIPs); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getGoogleRoutes(networkInterface string) ([]string, error) {
+	arguments := strings.Split(fmt.Sprintf("route list table local type local scope host dev %s proto 66", networkInterface), " ")
+	cmd := exec.Command("ip", arguments...)
+	b, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+	if len(b) == 0 {
+		return nil, fmt.Errorf("alias IPs not configured")
+	}
+	var res []string
+	for _, line := range strings.Split(string(b), "\n") {
+		ip := strings.Split(line, " ")
+		if len(ip) >= 2 {
+			res = append(res, ip[1])
+		}
+	}
+	return res, nil
+}
+
+func TestAliasAgentRestart(t *testing.T) {
+	var networkInterface string
+
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("couldn't get image from metadata")
+	}
+
+	switch {
+	case strings.Contains(image, "debian-10"), strings.Contains(image, "ubuntu"):
+		networkInterface = defaultPredictableInterface
+	default:
+		networkInterface = defaultInterface
+	}
+
+	beforeRestart, err := getGoogleRoutes(networkInterface)
+	if err != nil {
+		t.Fatal(err)
+	}
+	cmd := exec.Command("systemctl", "restart", "google-guest-agent")
+	_, err = cmd.Output()
+	if err != nil {
+		t.Fatal(err)
+	}
+	afterRestart, err := getGoogleRoutes(networkInterface)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !compare(beforeRestart, afterRestart) {
+		t.Fatalf("routes are inconsistent after restart, before %v after %v", beforeRestart, afterRestart)
+	}
+	if err := verifyIPExist(afterRestart); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func verifyIPExist(afterRestart []string) error {
+	expected, err := utils.GetMetadata("network-interfaces/0/ip-aliases/0")
+	if err != nil {
+		return fmt.Errorf("couldn't get image from metadata: %v", err)
+	}
+	for _, ip := range afterRestart {
+		if ip == expected {
```

Comment:
did you test this? i don't think the format from ip-aliases key usually will match the output from `ip` command; usually the metadata key has a mask suffix

Reply:
Yes, the metadata key value and `ip` command (`local 10.14.1.0/24`) both have the mask suffix